examples/qualcomm/llama2/llama_qaihub.py

```diff
@@ -0,0 +1,232 @@
+    generate_qnn_executorch_option,
+)
+from executorch.examples.qualcomm.scripts.utils import (
+    setup_common_args_and_variables,
+    SimpleADB,
+)
+from executorch.exir.backend.backend_api import to_backend
+from executorch.exir.passes.memory_planning_pass import MemoryPlanningPass
+
+
+if __name__ == "__main__":
+    parser = setup_common_args_and_variables()
+
+    parser.add_argument(
+        "-a",
+        "--artifact",
+        help="path for storing generated artifacts by this example. Default ./llama2_qai_hub",
+        default="./llama2_qai_hub",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--context_binaries",
+        help="path to context binaries generated from qai_hub",
+        required=True,
+    )
+
+    parser.add_argument(
+        "--use_prompt_processor",
+        help="tokens will be evaluated all at once",
+        default=False,
+        action="store_true",
+    )
+
+    parser.add_argument(
+        "--tokenizer_bin",
+        help="llama2 tokenizer binary",
+        required=True,
+        type=str,
+    )
+
+    parser.add_argument(
+        "--seq_len",
+        help="ouput sequence length for llama2",
+        default=128,
+        type=int,
+    )
+
+    parser.add_argument(
+        "--temperature",
+        help="sampling temperature for llama2",
+        default=0.8,
+        type=float,
+    )
+
+    parser.add_argument(
+        "--prompt",
+        help="user prompts for llama2",
+        required=True,
+        type=str,
+    )
+
+    parser.add_argument(
+        "--pre_gen_pte",
+        help="folder path to pre-compiled ptes",
+        default=None,
+        type=str,
+    )
+
+    args = parser.parse_args()
+    target_names = (
+        [
+            f"llama_v2_7b_chat_quantized_Llama2_PromptProcessor_{i}_Quantized.bin"
+            for i in range(1, 5)
+        ]
+        if args.use_prompt_processor
+        else [
+            f"llama_v2_7b_chat_quantized_Llama2_TokenGenerator_{i}_Quantized.bin"
+            for i in range(1, 5)
+        ]
+    )
+
+    # common part for compile & inference
+    backend_options = generate_htp_compiler_spec(
+        use_fp16=False,
+        use_multi_contexts=True,
+    )
+    compiler_specs = generate_qnn_executorch_compiler_spec(
+        soc_model=eval(f"QcomChipset.{args.model}"),
+        backend_options=backend_options,
+        is_from_context_binary=True,
+    )
+
+    if args.pre_gen_pte is None:
+        # create custom operators as context loader
+        bundle_programs = []
+        for i, target in enumerate(target_names):
+            file_name = f"{args.context_binaries}/{target}"
+            bundle_programs.append(from_context_binary(file_name, f"ctx_loader_{i}"))
+        # lower with QnnBackend
+        lowered_modules = []
+        for prog in bundle_programs:
+            lowered_modules.append(
+                to_backend("QnnBackend", prog["edge_program"], compiler_specs)
+            )
```

Comment:
nit: why not use a comprehension?
`lowered_models = [to_backend("QnnBackend", prog["edge_program"], compiler_specs for prog in bundle_programs]`

Reply:
Will do, thank you.